### ISSUE TITLE
Deadlock detected after pg_repack receives SIGINT

### ISSUE THREAD

**Issue by jirihlinka** [reorg/pg_repack#55]
I'm running a pg_repack from a bash script with timeout of 10 minutes like so (simplified version):
timeout -s SIGINT 10m pg_repack --table=frequently_updated_table

After the 10 min timeout, the OS sends SIGINT to pg_repack process so the pg_repack calls:
SELECT repack.repack_drop($1, $2)
and it causes a deadlock with other process which is INSERTing into frequently_updated_table that has a pg_repack's trigger to log every change to a repack.log_[OID of frequently_updated_table].

Is it a bug of pg_repack or am I doing something wrong? Should I add some detailed info?

Thank You for any help,
Jiri

Update: pg_repack version 1.3.2

**Comment by amitlan** [comment]
It's highly likely that it's a bug of repack.repack_drop().

Could you check if you the issue is resolved with 5317f527f4a502347782030e30ca05db83c57584?

**Comment by jirihlinka** [comment]
Thanks Amit for reply. May I kindly ask You for a help (or link to how-to
:-)) with how to try the 5317f52? I'm using RPM from yum.postgresql.org
currently (
http://yum.postgresql.org/9.2/redhat/rhel-6-x86_64/pg_repack92-1.3.2-1.rhel6.x86_64.rpm
).

On Fri, Nov 6, 2015 at 9:18 AM, Amit Langote <notifications@github.com>
wrote:

> It's highly likely that it's a bug of repack.repack_drop().
>
> Could you check if you the issue is resolved with 5317f52
> <https://github.com/reorg/pg_repack/commit/5317f527f4a502347782030e30ca05db83c57584>
> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/reorg/pg_repack/issues/55#issuecomment-154341817>.
>



-- 
Bc. Jiří Hlinka
Tel.: 725 315 263

**Comment by amitlan** [comment]
@jirihlinka you can clone the repository [here](https://github.com/amitlan/pg_repack.git) and build pg_repack from the branch issue-55-fix. Have you tried building pg_repack from source before? Something like below?

```
cd /tmp
git clone https://github.com/amitlan/pg_repack.git
cd pg_repack
git checkout issue-55-fix
make install
```

**Comment by jirihlinka** [comment]
@amitlan : Great, thanks for help.

Now i'm testing the fix. As I'm not able simulate the deadlock use case, I'm not able to 100% determine the fix helped, but I'll let run the pg_repack for a few days and if the deadlock will not raise, I guess its 80% sure the fix helped.

Or is there anything else I can do to help with this patch?

**Comment by amitlan** [comment]
Hey @jirihlinka, you could try to reproduce the deadlock using steps  I described in [this](https://github.com/reorg/pg_repack/pull/56#issuecomment-154419676) comment on the PR I sent to fix this issue.

**Comment by jirihlinka** [comment]
@amitlan : nice! I'll check it as soon as possible (tuesday / wednesday)

**Comment by schmiddy** [comment]
I'm hoping this issue is fixed with #56, thanks to @amitlan for chasing that down. Feel free to re-open if you still see the problem with git master.
